OpenRA.Mods.Common/Traits/Multipliers/VariedCostMultiplier.cs

```diff
@@ -0,0 +1,48 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Modifies the production cost of this actor for a specific queue or when a prerequisite is granted. " +
+		"Requires `VariedCostManager` on the world actor.")]
+	public class VariedCostMultiplierInfo : TraitInfo<VariedCostMultiplier>, IProductionCostModifierInfo, IRulesetLoaded
+	{
+		[Desc("Only apply this cost change if the owner has these prerequisites.")]
+		public readonly string[] Prerequisites = { };
+
+		[Desc("Production queues that this cost will apply to.")]
+		public readonly HashSet<string> Queues = new HashSet<string>();
+
+		[Desc("Set this if items should get the same random pricing.")]
+		public readonly string Group = null;
+
+		int IProductionCostModifierInfo.GetProductionCostModifier(World world, ActorInfo actorInfo, TechTree techTree, string queue)
+		{
+			if ((!Queues.Any() || Queues.Contains(queue)) && (!Prerequisites.Any() || techTree.HasPrerequisites(Prerequisites)))
```

Comment:
```suggestion
			if ((Queues.Count == 0 || Queues.Contains(queue)) && (Prerequisites.Count == 0 || techTree.HasPrerequisites(Prerequisites)))
```
Should make this both faster to read and faster to execute (I know it's the same in `ProductionCostMultiplierInfo`, but you can change it there as well).

Reply:
That eliminates LINQ completely.